test/e2e/capi_test.go

```diff
@@ -259,11 +259,12 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 						ControlPlaneWaiters: clusterctl.ControlPlaneWaiters{
 							WaitForControlPlaneInitialized: EnsureControlPlaneInitialized,
 						},
-						InitWithBinary:                  "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.5/clusterctl-{OS}-{ARCH}",
-						InitWithCoreProvider:            "cluster-api:v1.0.5",
-						InitWithBootstrapProviders:      []string{"kubeadm:v1.0.5"},
-						InitWithControlPlaneProviders:   []string{"kubeadm:v1.0.5"},
-						InitWithInfrastructureProviders: []string{"azure:v1.0.2"},
+						InitWithKubernetesVersion:       "v1.26.12",
+						InitWithBinary:                  "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.5.4/clusterctl-{OS}-{ARCH}",
+						InitWithCoreProvider:            "cluster-api:v1.5.4",
+						InitWithBootstrapProviders:      []string{"kubeadm:v1.5.4"},
+						InitWithControlPlaneProviders:   []string{"kubeadm:v1.5.4"},
+						InitWithInfrastructureProviders: []string{"azure:v1.11.7"},
```

Comment:
Is there a way we could fetch these values from `azure-dev.yaml` so that we do not maintain it going ahead? Not a blocking comment per se.

Reply:
since we're adding a second test with a different set of versions we need to configure this per test https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4446

We could have two separate sets of env variables 🤔 Maybe something to follow up on in #4446 @willie-yao? Let's merge this one as-is for now to fix the broken test

Reply:
I agree that having two sets of env variables would be the better way to go about this. Will work on implementing it for my PR!